winbuild/dir.cpp

```diff
@@ -48,22 +50,104 @@ class WinDirHandle : public watchman_dir_handle {
   }
 
   const watchman_dir_ent* readDir() override {
```

Comment:
would you mind breaking this up a little bit so that the win7 and win8+ versions are separate functions?
That will help keep the diff smaller; logically you'd be renaming the existing method and adding the helper from this comment, and another method for the win7 logic.

```
const watchman_dir_ent* readDir() override {
   if (win7_) {
      return readDirWin7();
   }
   return readDirWin8();
}
```